### PR TITLE
Additional tests [CN-1286]

### PR DESCRIPTION
**Pull request by SeriyBg**
The PR contains examples of possible tests that must be added to the test-cases

### CODE REVIEW

**Review by semihbkgr — APPROVED** [review]

File: internal/controller/hazelcast/mutate/mutate_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package mutate
+
+import (
+	"testing"
+
+	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	n "github.com/hazelcast/hazelcast-platform-operator/internal/naming"
+)
+
+func Test_HazelcastSpec_SetDefauktValues(t *testing.T) {
```
Comment (kutluhanmetin) [inline internal/controller/hazelcast/mutate/mutate_test.go:10]:
```suggestion
func Test_HazelcastSpec_SetDefaultValues(t *testing.T) {
```

File: internal/controller/hazelcast/mutate/mutate_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	h := &hazelcastv1alpha1.Hazelcast{
+		Spec: hazelcastv1alpha1.HazelcastSpec{
+			Repository:                 n.HazelcastRepo,
+			DeprecatedLicenseKeySecret: "my-license-key",
+		},
+	}
+	if mutated := HazelcastSpec(h); !mutated {
+		t.Errorf("Expecting Hazelcast Spec to be mutated to default values")
+	}
+	if h.Spec.Repository != n.HazelcastEERepo {
```
Comment (kutluhanmetin) [inline internal/controller/hazelcast/mutate/mutate_test.go:20]: Shall we add this as an expect case to the end of the test? expect(HazelcastEERepo)

**Review by kutluhanmetin — APPROVED** [review]
looks good!

File: internal/controller/hazelcast/mutate/mutate_test.go (hunk 2 above)
Comment (SeriyBg) [reply internal/controller/hazelcast/mutate/mutate_test.go:20]: What's the difference between its placing here and at the end of the test?

**Pull request merged**